map-rendering/app/src/main/java/com/here/android/example/map/basic/MainActivity.java

```diff
@@ -49,18 +50,48 @@ protected void onCreate(Bundle savedInstanceState) {
      */
     private void requestPermissions() {
 
-        final List<String> requiredSDKPermissions = new ArrayList<String>();
-        requiredSDKPermissions.add(Manifest.permission.ACCESS_FINE_LOCATION);
-        requiredSDKPermissions.add(Manifest.permission.WRITE_EXTERNAL_STORAGE);
-        requiredSDKPermissions.add(Manifest.permission.INTERNET);
-        requiredSDKPermissions.add(Manifest.permission.ACCESS_WIFI_STATE);
-        requiredSDKPermissions.add(Manifest.permission.ACCESS_NETWORK_STATE);
+        List<String> permissionList = Arrays.asList(
+                Manifest.permission.ACCESS_FINE_LOCATION,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE,
+                Manifest.permission.INTERNET,
+                Manifest.permission.ACCESS_WIFI_STATE,
+                Manifest.permission.ACCESS_NETWORK_STATE);
+
+        // check list of permissions
+        Boolean isPermissionsGranted = true;
+        for (String permission : permissionList) {
+            if (ContextCompat.checkSelfPermission(this, permission) != PackageManager.PERMISSION_GRANTED) {
+                isPermissionsGranted = false;
+                break;
+            }
+        }
+
+        // return if all permissions already granted
+        if (isPermissionsGranted) {
+            createMapFragmentView();
+            return;
+        }
+
+        // request permissions
+        if (android.os.Build.VERSION.SDK_INT >= 23){
```

Comment:
Do we need to check the Android version here? The targetSDKVersion in the build.gradle has been set to 23

Reply:
But the `minSdkVersion` is set to **16**.

Reply:
Yes, I think this is good as-is. We *could* replace the hard-coded `23` here with `Build.VERSION_CODES.M` like in PR #26, but personally I actually like seeing the number better than reading the code name.